### PR TITLE
Feature: App-739 Remove Action

### PR DESCRIPTION
**Pull request by Fabricevladimir**
## Description

- adds remove authorized wallets action to the proposal flow
- adds basic screen validation to Configure Actions screen
- fixes ability to continue with all whitespace proposal title & summary

**Note:** Allowing empty fields to go through as valid in order to mirror create dao user experience. Idea is to filter the list before submitting to SDK anyway. 

Also, reviewers, because #443 isn't merged yet and I needed a few components from that task, I had to merge the feature branch into this current one 😢 

Task: [APP-739](https://aragonassociation.atlassian.net/browse/APP-739)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

### CODE REVIEW

File: packages/web-app/src/pages/newProposal.tsx [hunk 1]
```diff
@@ -39,6 +41,7 @@ const NewProposal: React.FC = () => {
    *                    Render                     *
    *************************************************/
 
+  console.log(errors, dirtyFields);
```
Comment (WissenIstNacht) [inline packages/web-app/src/pages/newProposal.tsx]: remove?

File: packages/web-app/src/pages/newProposal.tsx [hunk 2]
```diff
@@ -72,6 +75,7 @@ const NewProposal: React.FC = () => {
             <Step
               wizardTitle={t('newProposal.configureActions.heading')}
               wizardDescription={t('newProposal.configureActions.description')}
+              isNextButtonDisabled={!actionsAreValid(errors)}
```
Comment (WissenIstNacht) [inline packages/web-app/src/pages/newProposal.tsx:79]: Thanks for improving this. However, I think it's still not quite correct. You can continue with empty but "undirtied" fields. I.e., when you add a fresh action.

File: packages/web-app/src/containers/manageWalletsModal/index.tsx [hunk 3]
```diff
@@ -46,37 +45,80 @@ const ManageWalletsModal: React.FC<ManageWalletsModalProps> = ({
...
+      const temp: SelectableWallets = {};
+      initialSelections.forEach(address => {
+        temp[address] = true;
+      });
+      setSelectedWallets(temp);
+    }
+  }, [initialSelections]);
 
+  // handles select all checkbox
+  const handleSelectAll = () => {
```
Comment (WissenIstNacht) [inline packages/web-app/src/containers/manageWalletsModal/index.tsx:84]: FYI: Assume you have n addresses total. You search `0xa`, reducing it to a subset of size k (<n). If you then apply `select all` you will end up selecting n addresses. I would expect to only select the k addresses returned by the search.

Not sure what the right implementation is. Up to you to change/leave it.

File: packages/web-app/src/containers/manageWalletsModal/index.tsx [hunk 4]
```diff
@@ -12,29 +12,28 @@ import ModalBottomSheetSwitcher from 'components/modalBottomSheetSwitcher';
...
-];
-
 type ManageWalletsModalProps = {
   addWalletCallback: (wallets: Array<string>) => void;
   resetOnClose?: boolean;
+  wallets: Array<string>;
+  initialSelections: Array<string>;
 };
 
+type SelectableWallets = Record<string, boolean>;
```
Comment (WissenIstNacht) [inline packages/web-app/src/containers/manageWalletsModal/index.tsx]: I don't want to change your style, so don't take this too seriously, but: This is really what a set does. In other words, the boolean assignment is kinda redundant, as the presence of the address in a set already singals that. :)

File: packages/web-app/src/containers/manageWalletsModal/index.tsx [hunk 5]
```diff
@@ -46,37 +45,80 @@ const ManageWalletsModal: React.FC<ManageWalletsModalProps> = ({
...
+    }
+  };
+
+  // handles checkbox selection for individual wallets
   const handleSelectWallet = (wallet: string) => {
     const tempSelectedWallets = {...selectedWallets};
     tempSelectedWallets[wallet]
       ? delete tempSelectedWallets[wallet]
       : (tempSelectedWallets[wallet] = true);
     setSelectedWallets(tempSelectedWallets);
```
Comment (WissenIstNacht) [inline packages/web-app/src/containers/manageWalletsModal/index.tsx]: When re-setting state based on previous state the correct approach is to do `setState(callBackThatTransformsPreviousToNextState)`. In this case: `setSelectedWallets((previous) => {const temp = {...previous}; // etc...}`

File: packages/web-app/src/containers/actionBuilder/addAddresses/emptyState.tsx [hunk 6]
```diff
@@ -10,14 +10,15 @@ type Props = {
   buttonLabel: string;
 };
 
+// This needs to be removed when the empty state is implemented
```
Comment (WissenIstNacht) [inline packages/web-app/src/containers/actionBuilder/addAddresses/emptyState.tsx:13]: Well, ackchually...

**Review by WissenIstNacht — CHANGES_REQUESTED** [review]
Thanks man, lots of good stuff.

File: packages/web-app/src/pages/newProposal.tsx (hunk 2 above)
Comment (Fabricevladimir) [reply packages/web-app/src/pages/newProposal.tsx:79]: Crap. I removed required validation actually to match the create DAO experience. The only issue is that it's the metadata resources & links not the wallets that don't take empty fields into account 🤦🏾‍♂️ 

Update: Apparently validation is getting overhauled. Opening a new ticket that will include this.

File: packages/web-app/src/containers/manageWalletsModal/index.tsx (hunk 3 above)
Comment (Fabricevladimir) [reply packages/web-app/src/containers/manageWalletsModal/index.tsx:84]: I'm tempted to create a refactor ticket for this. I adapted the modal to my needs, but I didn't exactly do a deep dive into the existing logic. In fact, even lifting the state might be an improvement.

File: packages/web-app/src/containers/manageWalletsModal/index.tsx (hunk 4 above)
Comment (Fabricevladimir) [reply packages/web-app/src/containers/manageWalletsModal/index.tsx]: Merely aliased the type, this was already built.

File: packages/web-app/src/containers/manageWalletsModal/index.tsx (hunk 5 above)
Comment (Fabricevladimir) [reply packages/web-app/src/containers/manageWalletsModal/index.tsx]: I didn't catch this, thanks

File: packages/web-app/src/containers/actionBuilder/addAddresses/emptyState.tsx (hunk 6 above)
Comment (Fabricevladimir) [reply packages/web-app/src/containers/actionBuilder/addAddresses/emptyState.tsx:13]: Yes, but it only supports IllustrationHuman as the main illustration. I opened a ticket for a refactor/extension for it.

**Review by WissenIstNacht — APPROVED** [review]
Yup, thanks for the changes 👍🏼

File: packages/web-app/src/containers/manageWalletsModal/index.tsx [hunk 7]
```diff
@@ -12,71 +12,119 @@ import ModalBottomSheetSwitcher from 'components/modalBottomSheetSwitcher';
...
+
+      if (previousState.has(wallet)) temp.delete(wallet);
+      else temp.add(wallet);
+
+      return temp;
+    });
   };
 
+  // handles cleanup after modal is closed.
+  // @RakeshUp considering the initialSelection state, is it
```
Comment (RakeshUP) [inline packages/web-app/src/containers/manageWalletsModal/index.tsx]: we can remove resetOnClose if the parent component is going to handle transitory state

**Review by RakeshUP — APPROVED** [review]

**Pull request merged**